### PR TITLE
Finish grpc create app

### PR DESCRIPTION
**Pull request by aguerra**
For now I'm postponing error translation and setting the defaults on the server as this PR is already too big.

Related #203.

---
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/207)

### CODE REVIEW

File: cmd/client/cmd/app.go [hunk 1]
```diff
@@ -42,85 +47,124 @@ The app name must follow this rules:
...
+
+	team, err := cmd.Flags().GetString("team")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Invalid team parameter: ", err)
+		return
+	}
+	if team == "" {
+		fmt.Fprintln(os.Stderr, "You must provide the name of the team")
+		return
+	}
```
Comment (drgarcia1986) [inline cmd/client/cmd/app.go]: what about simplify this validation, like this:
```go
team, err := cmd.Flags().GetString("team")
if err != nil || team == "" {
    fmt.Fprintln(os.Stderr, "You must provide a valid name of the team")
    return
}
```
?
(or just ignore error and only check for blank team name)

File: cmd/client/cmd/app.go [hunk 2]
```diff
@@ -42,85 +47,124 @@ The app name must follow this rules:
...
+
+	scaleMax, err := cmd.Flags().GetInt32("scale-max")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Invalid scale-max parameter: ", err)
+		return
+	}
+
+	scaleMin, _ := cmd.Flags().GetInt32("scale-min")
+	if err != nil {
```
Comment (drgarcia1986) [inline cmd/client/cmd/app.go]:
```diff
- scaleMin, _ := cmd.Flags().GetInt32("scale-min")
+ scaleMin, err := cmd.Flags().GetInt32("scale-min")
```

File: pkg/server/app/models_test.go [hunk 3]
```diff
@@ -86,9 +86,9 @@ func newCreateRequest() *appb.CreateRequest {
 		},
 	}
 	as := &appb.CreateRequest_AutoScale{
-		CpuTargetUtilization: int64(42),
-		Max:                  int64(666),
-		Min:                  int64(1),
+		CpuTargetUtilization: 42,
+		Max:                  666,
```
Comment (drgarcia1986) [inline pkg/server/app/models_test.go]: 👹

File: pkg/server/k8s/client.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+type k8sClient struct {
+	kc *kubernetes.Clientset
+}
+
+func newNs(a *app.App, user string) *k8sv1.Namespace {
+	return &k8sv1.Namespace{
+		ObjectMeta: k8sv1.ObjectMeta{
+			Name: a.Name,
+			Labels: map[string]string{
+				"teresa.io/team": a.Team,
```
Comment (drgarcia1986) [inline pkg/server/k8s/client.go]: Maybe I found a bug on #205 .
I was getting app team for `teresa.io/app` annotation 😕 .

Comment (aguerra) [inline pkg/server/k8s/client.go]: We talked and settled for using the label.

File: cmd/client/cmd/app.go (hunk 1 above)
Comment (aguerra) [inline cmd/client/cmd/app.go]: Your snippet is better.

File: pkg/server/k8s/client.go (hunk 4 above)
Comment (drgarcia1986) [inline pkg/server/k8s/client.go]: BTW, fixed

File: pkg/server/app/app.go [hunk 5]
```diff
@@ -20,11 +20,14 @@ type Operations interface {
...
 type K8sOperations interface {
-	Create(app *App, st st.Storage) error
 	NamespaceAnnotation(namespace, annotation string) (string, error)
 	NamespaceLabel(namespace, label string) (string, error)
 	PodList(namespace string) ([]*Pod, error)
 	PodLogs(namespace, podName string, lines int64, follow bool) (io.ReadCloser, error)
+	CreateNamespace(*App, string) error
+	CreateQuota(*App) error
+	CreateSecret(string, string, map[string][]byte) error
+	CreateAutoScale(*App) error
```
Comment (drgarcia1986) [inline pkg/server/app/app.go]: 💅 maybe a simple name for these args helps to understand they proposal 💅

File: pkg/server/app/app_test.go [hunk 6]
```diff
@@ -168,3 +198,57 @@ func TestAppOperationsLogsErrNotFound(t *testing.T) {
...
+	app := &App{Name: "teresa", Team: name}
+	tops.(*team.FakeOperations).Storage[name] = &storage.Team{
+		Name:  name,
+		Users: []storage.User{*user},
+	}
+	ops.(*AppOperations).kops = &errK8sOperations{SecretErr: errors.New("Secret Error")}
+
+	if err := ops.Create(user, app); err == nil {
+		t.Errorf("expected error, got nil")
+	}
```
Comment (drgarcia1986) [inline pkg/server/app/app_test.go]:
```diff
- if err := ops.Create(user, app); err == nil {
+ if ops.Create(user, app) == nil {
```

File: pkg/server/app/app_test.go [hunk 7]
```diff
@@ -168,3 +198,57 @@ func TestAppOperationsLogsErrNotFound(t *testing.T) {
...
+	name := "luizalabs"
+	user := &storage.User{Email: "teresa@luizalabs.com"}
+	app := &App{Name: "teresa", Team: name}
+	tops.(*team.FakeOperations).Storage[name] = &storage.Team{
+		Name:  name,
+		Users: []storage.User{*user},
+	}
+	ops.(*AppOperations).kops = &errK8sOperations{QuotaErr: errors.New("Quota Error")}
+
+	if err := ops.Create(user, app); err == nil {
```
Comment (drgarcia1986) [inline pkg/server/app/app_test.go]:
```diff
- if err := ops.Create(user, app); err == nil {
+ if ops.Create(user, app) == nil {
```

File: pkg/server/app/app_test.go [hunk 8]
```diff
@@ -168,3 +198,57 @@ func TestAppOperationsLogsErrNotFound(t *testing.T) {
...
+	name := "luizalabs"
+	user := &storage.User{Email: "teresa@luizalabs.com"}
+	app := &App{Name: "teresa", Team: name}
+	tops.(*team.FakeOperations).Storage[name] = &storage.Team{
+		Name:  name,
+		Users: []storage.User{*user},
+	}
+	ops.(*AppOperations).kops = &errK8sOperations{AutoScaleErr: errors.New("AutoScale Error")}
+
+	if err := ops.Create(user, app); err == nil {
```
Comment (drgarcia1986) [inline pkg/server/app/app_test.go]:
```diff
- if err := ops.Create(user, app); err == nil {
+ if ops.Create(user, app) == nil {
```

**Pull request merged**